### PR TITLE
JBPM-6880 - messageRef property switches from name to message id upon…

### PR DESCRIPTION
**Pull request by tsurdilo**
… process save

### CODE REVIEW

**Comment by tsurdilo** [comment]
@hasys do you mind taking a look? thanks.

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshallerTest.java [hunk 1]
```diff
@@ -1008,4 +1008,35 @@ public void testBoundaryEventDefinitionRef() throws Exception {
         assertNotNull(endEvent);
     }
 
+    @Test
+    public void testMessageRefNodes() throws Exception {
```
Comment (jomarko) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshallerTest.java]: @tsurdilo could you please have a look on formatting in this test? Especially `getChildByName` calls seems to be not well formatted.

**Comment by tsurdilo** [comment]
@hasys ok formatting applied and updated to use util method. thanks.

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by tsurdilo** [comment]
Jenkins execute full downstream build

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshaller.java [hunk 2]
```diff
@@ -742,9 +742,7 @@ private void setCatchEventProperties(CatchEvent event,
                 }
             } else if (ed instanceof MessageEventDefinition) {
                 if (((MessageEventDefinition) ed).getMessageRef() != null) {
```
Comment (jomarko) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshaller.java]: We don't need this `if`, as it is done inside `setMessageRefProperties` implementation.

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshaller.java [hunk 3]
```diff
@@ -899,9 +910,7 @@ private void setThrowEventProperties(ThrowEvent event,
...
```
Comment (jomarko) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshaller.java]: We don't need this `if`, as it is done inside `setMessageRefProperties` implementation.

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshaller.java [hunk 4]
```diff
@@ -1897,15 +1906,13 @@ protected void marshallTask(Task task,
             taskType = "Send";
             SendTask st = (SendTask) task;
             if (st.getMessageRef() != null) {
```
Comment (jomarko) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshaller.java]: We don't need this `if`, as it is done inside `setMessageRefProperties` implementation.

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshaller.java [hunk 5]
```diff
@@ -1897,15 +1906,13 @@ protected void marshallTask(Task task,
...
-                properties.put("messageref",
-                               st.getMessageRef().getId());
+                setMessageRefProperties(properties, st.getMessageRef());
             }
         } else if (task instanceof ReceiveTask) {
             taskType = "Receive";
             ReceiveTask rt = (ReceiveTask) task;
             if (rt.getMessageRef() != null) {
```
Comment (jomarko) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshaller.java]: We don't need this `if`, as it is done inside `setMessageRefProperties` implementation.

**Pull request merged**